tap-agent/src/agent/sender_account.rs

```diff
@@ -509,16 +508,17 @@ impl Actor for SenderAccount {
 
         let myself_clone = myself.clone();
         let pgpool_clone = pgpool.clone();
-        let _escrow_account_monitor = escrow_accounts.clone().pipe_async(move |escrow_account| {
-            let myself = myself_clone.clone();
-            let pgpool = pgpool_clone.clone();
-            // get balance or default value for sender
-            // this balance already takes into account thawing
-            let balance = escrow_account
-                .get_balance_for_sender(&sender_id)
-                .unwrap_or_default();
-
-            async move {
+        let mut accounts_clone = escrow_accounts.clone();
+        let _escrow_account_monitor = tokio::spawn(async move {
+            while accounts_clone.changed().await.is_ok() {
+                let escrow_account = accounts_clone.borrow().clone();
+                let myself = myself_clone.clone();
+                let pgpool = pgpool_clone.clone();
+                // Get balance or default value for sender
+                // this balance already takes into account thawing
+                let balance = escrow_account
+                    .get_balance_for_sender(&sender_id)
+                    .unwrap_or_default();
```

Comment:
I think `pipe_async` could become a function in our `watcher.rs` and be reused in multiple places.

Reply:
You don't need to do it on this PR. Just a follow up issue to our refactor

Reply:
yeah, satisfaction level of getting to cleaner code is just[🚀]

Reply:
Also can I create that function?

Reply:
Just created #456, send a message there so I can assign it to you.